Library/Homebrew/dev-cmd/bump.rb

```diff
@@ -497,4 +501,38 @@ def retrieve_and_display_info_and_open_pr(formula_or_cask, name, repositories, a
 
     system HOMEBREW_BREW_FILE, *bump_cask_pr_args
   end
+
+  sig { params(formula_or_cask: T.any(Formula, Cask::Cask)).returns(T::Boolean) }
+  def synced_with_other_formulae?(formula_or_cask)
+    return false if formula_or_cask.is_a?(Cask)
+
+    synced_versions_formulae_file = "#{formula_or_cask.tap.path}/synced_versions_formulae.json"
+    return false unless File.exist?(synced_versions_formulae_file)
+
+    synced_versions_formulae = JSON.parse(File.read(synced_versions_formulae_file))
+    synced_versions_formulae.any? { |synced_formulae| synced_formulae.include?(formula_or_cask.name) }
+  end
+
+  sig {
+    params(
+      formula:     Formula,
+      new_version: T.nilable(T.any(Version, Cask::DSL::Version)),
+    ).returns(T::Array[String])
+  }
+  def synced_with(formula, new_version)
+    synced_with = []
+
+    JSON.parse(File.read("#{T.must(formula.tap).path}/synced_versions_formulae.json")).each do |synced_formulae|
```

Comment:
Similarly might want to cache this read/parse into a separate method called in both places.